### PR TITLE
Update shopify app gem

### PR DESCRIPTION
**Pull request by kevinhughes27**
This updates the embedded app example to use the latest `shopify_app` gem. The new gem is a proper rails engine and thus allows for real code sharing (you'll see that lots of code has been replaced with a including a concern from the gem). 

I :tophat:ed locally and on a new heroku app. The latter required that I run the `shop_model` generator because Heroku really doesn't play nice with the InMemorySession (not even for a single request). I'll leave the shop_model commit for review purposes and we can discuss if the example should contain it or not before shipping.

@garymardell @kmcphillips

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,50 +1,3 @@
...
-    return unless domain = ShopifyApp.configuration.myshopify_domain.presence || "myshopify.com"
-
-    name = params[:shop].to_s.strip
-    name += ".#{domain}" if !name.include?(domain) && !name.include?(".")
-    name.sub!(%r|https?://|, '')
-
-    u = URI("http://#{name}")
-    u.host.ends_with?(".#{domain}") ? u.host : nil
-  end
+  include ShopifyApp::SessionsController
```
Comment (celsodantas) [inline app/controllers/sessions_controller.rb]: <3

**Comment by celsodantas** [comment]
:+1:

**Comment by kevinhughes27** [comment]
also should have referenced the PR to shopify_app here https://github.com/Shopify/shopify_app/pull/124

**Comment by garymardell** [comment]
:+1:

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (kmcphillips) [inline app/controllers/sessions_controller.rb]: OMG

**Comment by kmcphillips** [comment]
Code here looks good. Code in Shopify/shopify_app#124 is amazing. So good.

**Comment by kevinhughes27** [comment]
thanks @kmcphillips!

**Pull request merged**